warehouse/integrations/snowflake/snowflake.go

```diff
@@ -815,6 +820,16 @@ func (sf *Snowflake) LoadIdentityMappingsTable(ctx context.Context) error {
 	return nil
 }
 
+// ShouldAppend returns true if the load table strategy is append mode and the source is event stream
+// currently, source category can either be (cloud, singer-protocol, warehouse, webhook)
+func (sf *Snowflake) ShouldAppend() bool {
+	sourceCategory := sf.Warehouse.Source.SourceDefinition.Category
+	if _, isAppendCategory := appendSourceCategoryMap[sourceCategory]; isAppendCategory {
```

Comment:
If it is an append category then `isAppendCategory` is `true`, right? Then why do we return `false`?

I was expecting something like:

```go
func (sf *Snowflake) ShouldAppend() bool {
	sourceCategory := sf.Warehouse.Source.SourceDefinition.Category
	isAppendCategory := appendSourceCategoryMap[sourceCategory]

	return isAppendCategory && sf.config.loadTableStrategy == loadTableStrategyAppendMode
```

This would contradict what we're trying to do though which is to always `MERGE` if the source category is "cloud extract". So maybe the global `map` should be renamed from `appendSourceCategoryMap` to `mergeSourceCategoryMap` and if the warehouse source matches a key in that map we **do not append**. Correct?

Reply:
Sorry, I created an inverse map but failed to update the variable.